src/ddscxx/src/org/eclipse/cyclonedds/sub/qos/DataReaderQosDelegate.cpp

```diff
@@ -249,18 +301,29 @@ DataReaderQosDelegate::operator==(const DataReaderQosDelegate& other) const
 DataReaderQosDelegate&
 DataReaderQosDelegate::operator =(const org::eclipse::cyclonedds::topic::qos::TopicQosDelegate& tqos)
 {
-    durability_  = tqos.policy<dds::core::policy::Durability>();
-    deadline_    = tqos.policy<dds::core::policy::Deadline>();
-    budget_      = tqos.policy<dds::core::policy::LatencyBudget>();
-    liveliness_  = tqos.policy<dds::core::policy::Liveliness>();
-    reliability_ = tqos.policy<dds::core::policy::Reliability>();
-    order_       = tqos.policy<dds::core::policy::DestinationOrder>();
-    history_     = tqos.policy<dds::core::policy::History>();
-    resources_   = tqos.policy<dds::core::policy::ResourceLimits>();
-    ownership_   = tqos.policy<dds::core::policy::Ownership>();
+    if (tqos.present() & QP_DURABILITY)
```

Comment:
This should be looking at the `DURABILITY` flag for all settings

Reply:
OUCH!!!
I will fix this immediately!!!